src/js/FileSystem.js

```diff
@@ -77,27 +77,28 @@ class FileSystem {
         const fileHandle = file._fileHandle;
 
         const fileReader = await fileHandle.getFile();
-        return await fileReader.text();
+        return fileReader.text();
     }
 
     async readFileAsBlob(file) {
         const fileHandle = file._fileHandle;
 
-        return await fileHandle.getFile();
+        return fileHandle.getFile();
     }
 
-    async openFile(file) {
+    async openFile(file, append) {
         const fileHandle = file._fileHandle;
 
-        return await fileHandle.createWritable();
+        const options = { keepExistingData: append || false };
```

Comment:
When using append don't we need to set the position in the file?

```bash
async function append_to_file(data) {
    const file = await openedFile.getFile();
    const blob = new Blob([`${data}\n`], {type: 'text/plain'});

    const writable = await openedFile.createWritable({ keepExisting: true });
    await writable.write({ type: 'seek', position: file.size });
    await writable.write({ type: 'write', data: blob });
}
```

Reply:
Please fix remaining sonar issues

Reply:
Good catch. I think is better to simply remove this parameter, we are not using it. In this way it will be more simple. We can add it in a future if we see some feature that needs it.
Is there some feature in this tethered logging that needs it? I was thinking about stopping and continuing in the logging, but the user can change the data to record, that implies the header, etc. and that breaks the functionality.

Reply:
Previous the `fileEntry` was stored and retrieved for new sessions to allow users to append to an existing log.
For single sessions it is not required.

Reply:
Yes, but we can't do that now. We need the user to "explicitly" give permissions to the file using the file picker. We can't write into a file without this first. For this reason I removed this part of the code.

Reply:
For now a single session seems sufficient.

Reply:
I think I fixed the issues but for some reason Sonar has not validated it again...

Reply:
Rewriting some functions from async to promises in FileSystem class will need changes in other places too.